query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/converters/DefaultColumnRequestConverter.java

```diff
@@ -207,6 +211,63 @@ private boolean handleConversionForNullOrEmptyBytesLiteral(
     return false;
   }
 
+  private boolean isFilterForArrayColumnType(Filter filter, ColumnRequestContext context) {
+    return isSimpleAttributeExpression(filter.getLhs())
+        && filter.getRhs().getValueCase().equals(LITERAL)
+        && context.isArrayColumnType();
+  }
+
+  /** Handles value conversion of a array expression based */
+  private void handleConversionForArrayColumnExpression(
+      String lhs, String operator, Expression rhs, StringBuilder builder, Builder paramsBuilder) {
+    Value value = rhs.getLiteral().getValue();
+
+    if (handleConversionForNullOrEmptyArrayLiteral(lhs, operator, builder, value)) return;
+
+    builder.append(lhs);
+    // support only equals and IN operator
+    // both of them are handled as contains check
+    if (!operator.equals("=") && !operator.equals("IN")) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Unsupported operator {%s} for array column with non-empty value", operator));
+    } else {
+      // overlap operator for array
+      builder.append(" && ");
```

Comment:
Can we add comments that `IN` and `EQ` are treated as contains in existing impl (say pinot), and so in postgres it translated to same using `&&` operator - https://www.postgresql.org/docs/current/functions-array.html?